NewDawn/NewDawn/ProfileViewController/Profile_DraftFinal.swift

```diff
@@ -24,16 +24,126 @@ class Profile_DraftFinal: UIViewController {
     
     @IBAction func getStartedButtonTapped(_ sender: Any) {
         // TODO: Send all info to backend and go to profile page
+        let activityIndicator = self.prepareActivityIndicator()
+        
+        let request = createRegistrationRequest()
+        
+        if request == nil {
+            return
+        }
+        self.removeActivityIndicator(activityIndicator: activityIndicator)
+        self.processSessionTasks(request: request!, callback: readRegistrationResponse)
+        
     }
     
-    /*
-    // MARK: - Navigation
-
-    // In a storyboard-based application, you will often want to do a little preparation before navigation
-    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
-        // Get the new view controller using segue.destination.
-        // Pass the selected object to the new view controller.
+    func createRegistrationRequest() -> URLRequest? {
+        
+        let url = getURL(path: "register/")
+        var request = URLRequest(url:url)
+        request.httpMethod = "POST"
+        request.addValue("application/json", forHTTPHeaderField: "Content-Type")
+        request.addValue("application/json", forHTTPHeaderField: "Accept")
+        let register_info: [String: Any] = getUserInputInfo()
+        do {
+            request.httpBody = try JSONSerialization.data(withJSONObject: register_info, options: .prettyPrinted)
+        } catch let error {
+            // TODO: Sepearte error messages into engineer's and user's
+            print(error.localizedDescription)
+            self.displayMessage(userMessage: "Registration Request Creation Failed", dismiss: false)
+            return nil
+        }
+        return request
+    }
+    
+    func getUserInputInfo() -> [String: Any]{
```

Comment:
This function is too long. Split the code inside to functions like getUserData(), getAccountData(), getProfileData(), etc.

Reply:
And call these functions inside this one